### DISCUSSION TITLE
database "vaultwarden" has a collation version mismatch

### DISCUSSION THREAD

**Discussion by Stunt0265** [dani-garcia/vaultwarden#3658] [Q&A]
Running the latest version of Vaultwarden on Docker on Synology NAS.  Using latest version of Postgres, 15.3-1.pgdg120+1.  

Just started getting this today:

WARNING: database "vaultwarden" has a collation version mismatch DETAIL: The database was created using collation version 2.31, but the operating system provides version 2.36. HINT: Rebuild all objects in this database that use the default collation and run ALTER DATABASE vaultwarden REFRESH COLLATION VERSION, or build PostgreSQL with the right library version.

**Comment by tessus** [comment]
~https://github.com/dani-garcia/vaultwarden/wiki/Using-the-MariaDB-%28MySQL%29-Backend#foreign-key-errors-collation-and-charset~

Edit: Scratch that. You are not using MySQL.

**Comment by BlackDex** [comment]
I actually doubt it's a Vaultwarden issue. You probably updated your OS which caused this. The message also looks like it's coming from the PostgreSQL server.

I suggest to read the following https://www.postgresql.org/docs/current/sql-altercollation.html

**Comment by Stunt0265** [comment]
Running this on a NAS and the only OS upgrade would have been Synology OS at some point.  And maybe the Postgres docker container.

**Comment by BlackDex** [comment]
I would guess it would have been the PostgreSQL Docker container then. It probably now uses newer library versions and for that you need to do some manual database upgrading commands.

**Comment by pparedes1** [comment]
I had to login into my postgres database and run the command mentioned in the error to resolve.
1) `docker exec -it postgres /bin/bash` where postgres is your container name (if you are running locally not required)
2) `/bin/psql  --username=postgres_username` where postgres_username is your user for the database
3) `ALTER DATABASE vaultwarden_db REFRESH COLLATION VERSION;` output was NOTICE:  changing version from 2.31 to 2.36
4) `exit` to close session and `exit` to close access to container

**Reply by v11t** [reply]
for folks running vaultwarden in docker on synology, changing the second command to
`/bin/psql  --username=vaultwardenuser --db=vaultwardendb`
did the job for me (not specifying the db name threw an error)

**Comment by Stunt0265** [comment]
I've tried this multiple times and get no output.  Just returns to cursor.  I found a similar thread yesterday about Nextcloud.

**Turns out I was missing the semicolon at the end.  Duh.  Thanks for the help.

**Reply by onjour** [reply]
@Stunt0265 Did you follow Marius' hosting website instructions to build your Vaultwarden Portainer Stack? 

If that's the case, did you also ran into issues when Portainer pulled postgres 16? My database became invalid and I had to revert the container to 15. Now I'm also having a collation mismatch but at least Vaultwarden works again

I'd be interested in knowing your experience with this. Thank you

**Reply by pparedes1** [reply]
@onjour I was able to run `ALTER DATABASE vaultwarden_db REFRESH COLLATION VERSION;` on postgres 15 and received a response of NOTICE:  changing version from 2.31 to 2.36

Normally for my postgres containers I specify the version of postgres as major upgrades can break applications (e.g. version 15 to version 16 will likely break stuff). 

If you want information on how to upgrade  a docker database I relied heavily on [https://www.cloudytuts.com/tutorials/docker/how-to-upgrade-postgresql-in-docker-and-kubernetes/](url). I've gotten it down to a science where I bring up the new database version, created a script to upgrade and transfer the data between the old and the new, then rename the data folders to make sure all old applications still work. Hope that helps,
